dune_client/query.py

```diff
@@ -33,3 +33,10 @@ def url(self) -> str:
                 [self.base_url(), urllib.parse.quote_plus(params, safe="=&?")]
             )
         return self.base_url()
+
+    def __hash__(self) -> int:
+        """
+        This contains the query ID and the values of relevant parameters.
+        Thus, it is unique for caching purposes
+        """
+        return self.url().__hash__()
```

Comment:
Like this idea for hashing!

Reply:
Unfortunately this did not work out as I had expected:

https://github.com/cowprotocol/dune-client/issues/32